ladybug/hourlyplot.py

```diff
@@ -0,0 +1,424 @@
+        assert isinstance(base_point, Point3D), 'Expected Point3D for ' \
+            'HourlyPlot base point. Got {}.'.format(type(base_point))
+
+        # ensure the analysis period of the data collection has been validated
+        if not data_collection.validated_a_period:
+            data_collection = data_collection.validate_analysis_period()
+
+        # assign the inputs as properties of this data collection
+        self._data_collection = data_collection.to_immutable()
+        self._base_point = base_point
+        self._x_dim = self._check_dim(x_dim, 'x_dim')
+        self._y_dim = self._check_dim(y_dim, 'y_dim')
+        self._z_dim = 0
+        if z_dim != 0:
+            self._z_dim = self._check_dim(z_dim, 'z_dim')
+
+        # set properties to None that will be computed later
+        self._colored_mesh2d = None
+        self._colored_mesh3d = None
+        self._hour_points = None
+        self._hour_text = None
+        self._month_points = None
+        self._month_label_points = None
+        self._month_text = None
+
+        # create the graphic container from the inputs
+        a_per = self.analysis_period
+        if not a_per.is_reversed:
+            self._num_x = a_per.end_time.doy - a_per.st_time.doy + 1
+        else:
+            num_doys = 365 if not a_per.is_leap_year else 366
+            self._num_x = num_doys - a_per.st_time.doy + a_per.end_time.doy + 1
+        if a_per.end_hour == 23:  # the last datetime is not included
+            self._num_y = ((a_per.end_hour - a_per.st_hour + 1) * a_per.timestep)
+        else:  # the last datetime is included
+            self._num_y = ((a_per.end_hour - a_per.st_hour) * a_per.timestep) + 1
+        x_dist = x_dim * self._num_x
+        y_dist = y_dim * self._num_y
+        max_pt = Point3D(base_point.x + x_dist, base_point.y + y_dist, base_point.z)
+        self._container = GraphicContainer(
+            data_collection.values, base_point, max_pt, legend_parameters,
+            data_collection.header.data_type, data_collection.header.unit)
+
+    @property
+    def data_collection(self):
+        """The data collection assigned to this hourly plot."""
+        return self._data_collection
+
+    @property
+    def legend_parameters(self):
+        """The legend parameters customizing this hourly plot."""
+        return self._container.legend_parameters
+
+    @property
+    def base_point(self):
+        """Point3D for the base point of this hourly plot."""
+        return self._base_point
+
+    @property
+    def x_dim(self):
+        """A number for the X dimension of each cell of the hourly plot."""
+        return self._x_dim
+
+    @property
+    def y_dim(self):
+        """A number for the Y dimension of each cell of the hourly plot."""
+        return self._y_dim
+
+    @property
+    def z_dim(self):
+        """A number for the Z dimension of each cell of the hourly plot."""
+        return self._z_dim
```

Comment:
Not that it should be like that but I was expecting to be able to set these values without creating a new chart. Basically if the data stays the same plot should give me the ability to play around with these input values to get the right visualization without recreating the object.

Curious to know the reasoning behind making them read-only?

Reply:
The main reason for it is because everything you see running in `__init__` would have to be re-run every time that one of these properties changes. This includes generating a new graphic container object based on new min and max points that have now changed because the dimensions have changed.  It also prevents me from caching certain hidden variables that get reused in multiple properties.

Bottom line is that it's just a lot of extra code to enable flexibility that I'm not sure we're actually going to use. If we find situations that we really want this feature, then we can sit down and add in all of the extra code to enable it. But it just didn't seem worth it right now.